Update: Collapse Modal Sections When Modal is Disabled

jennsun opened the pull request:
https://app.clickup.com/t/xq9a7r

Updated initial SDK screen so that when either the “Unsaved Changes Confirmation Modal” or “Complete Video Confirmation Modal” are unchecked, the corresponding form fields are hidden.

jennsun committed:
Update: Collapse Modal Sections when Modal is Disabled

simanthastew commented on example_site/src/components/ConfigurationControls.js:
```diff
@@ -49,6 +49,132 @@ export default function ConfigurationControls({ isOpen }) {
+      </div>
+    );
+  }
+  //testing area
```
Make sure to remove errant comments!

simanthastew commented on example_site/src/components/ConfigurationControls.js:
```diff
@@ -49,6 +49,132 @@ export default function ConfigurationControls({ isOpen }) {
     shouldShowConfirmCompleteVideoModal,
   } = watchFields;
 
+  function renderUnsavedChanges() {
```
I think a better name for this function would be `renderUnsavedChangesModalForm`. 

There is definitely an art to variable naming, it's often touted as one of the toughest things to do! [This article](https://towardsdatascience.com/data-scientists-your-variable-names-are-awful-heres-how-to-fix-them-89053d2855be) is focusing on data science/python, but I thought it had some really nice general rules. My favorite bit is, "_Your code will be read more times than it is written._".  

When I read the original name, `renderUnsavedChanges`, I think there are "unsaved changes" that need to be rendered. If you add `ModalForm` to the end, it's a little more descriptive about what this function actually does.

simanthastew commented on example_site/src/components/ConfigurationControls.js:
```diff
@@ -49,6 +49,132 @@ export default function ConfigurationControls({ isOpen }) {
+    );
+  }
+
+  function renderCompleteVideo() {
```
Same comment as above, `renderCompleteVideo` --> `renderCompleteVideoModalForm`.

simanthastew commented on example_site/src/components/ConfigurationControls.js:
```diff
@@ -282,61 +408,11 @@ export default function ConfigurationControls({ isOpen }) {
-            defaultValue="Exit Editor"
-            ref={register}
-          />
+          { 
+            //Collapse Unsaved Changes Confirmation Modal
+            shouldShowUnsavedChangesModal && renderUnsavedChanges() 
+          }
```
I know they use this format in the React docs, but check out [this article](https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx) from Kent Dodds about the dangers of using `&&` instead of ternaries while rendering. The basic idea is that when you use a ternary, you get you be explicit about what you want to render when your variable evaluates to `falsey`, and gives some more details about what React is actually doing when it evaluates these statements.

simanthastew replied:
Additionally, you can conditionally render the HTML here instead of defining a function above, like so:
```
{shouldShowUnsavedChangesModal ? (
    <input>....</input>
) : (null }
```

simanthastew reviewed (changes requested):
Nice job! There are a few different ways we can restructure that I'd like to go over, and then let's talk about CSS transitions and other ways we can accomplish hiding/showing the forms when we chat later.

jennsun committed:
Update: PR Review
Removing errant comments, using conditional rendering based on ternaries rather than &&, removing render modal functions and placing HTML inside conditional render

jennsun committed:
Update: Change layout of SDK front page
Moved Labels portion underneath Editor, Created Modals header with Unsaved Changes and Complete Video Confirmation Modals to two separate divs

jennsun committed:
Update: Fade In/Out when Modal Checkbox checked/unchecked
Also reorganized code so conditional render of modal settings was based off ternary of opacity and transitioned the change in CSS

simanthastew commented on example_site/src/components/ConfigurationControls.js:
```diff
@@ -281,64 +284,80 @@ export default function ConfigurationControls({ isOpen }) {
+      </div>
+      
+      <div>
+        <h1></h1>
+        <h2></h2>
+        <br></br>
```
What are these additional elements for?

jennsun replied:
Intended for spacing, made changes so top padding (CSS) is used instead!

simanthastew reviewed (changes requested):
Looks great - just one question about a few extra DOM elements and we're good to go!
